scripts/src/java/org/oppia/android/scripts/maven/GenerateMavenDependenciesList.kt

```diff
@@ -0,0 +1,351 @@
+      TextFormat.printer().print(mavenDependencyList, writer)
+    }
+  }
+
+  private fun retrieveDependencyListFromPom(
+    finalDependenciesList: List<MavenListDependency>
+  ): MavenDependencyList {
+    val mavenDependencyList = mutableListOf<MavenDependency>()
+    finalDependenciesList.forEach {
+      // Remove ".jar" or ".aar" or any other extension from the specified url.
+      val pomFileUrl = "${it.url?.dropLast(3)}pom"
+      val artifactName = it.coord
+      val artifactVersion = StringBuilder()
+      var lastIndex = artifactName.length - 1
+      while (lastIndex >= 0 && artifactName[lastIndex] != ':') {
+        artifactVersion.append(artifactName[lastIndex])
+        lastIndex--
+      }
+      artifactVersion.reverse()
+      val pomFile = licenseFetcher.scrapeText(pomFileUrl)
+      val mavenDependency = MavenDependency.newBuilder().apply {
+        this.artifactName = it.coord
+        this.artifactVersion = artifactVersion.toString()
+        this.addAllLicense(extractLicenseLinksFromPom(pomFile))
+      }
+      mavenDependencyList.add(mavenDependency.build())
+    }
+    return MavenDependencyList.newBuilder().addAllMavenDependency(mavenDependencyList).build()
+  }
+
+  private fun retrieveThirdPartyMavenDependenciesList(
+    rootPath: String
+  ): List<String> {
+    return BazelClient(File(rootPath).absoluteFile, commandExecutor)
+      .retrieveThirdPartyMavenDepsListForBinary("oppia")
+      .map { dep ->
+        if (dep.startsWith(MAVEN_PREFIX)) dep.substring(MAVEN_PREFIX.length, dep.length) else dep
+      }
+  }
+
+  private fun getDependencyListFromMavenInstall(
+    pathToMavenInstall: String,
+    bazelQueryDepsNames: List<String>
+  ): List<MavenListDependency> {
+    val mavenInstallJsonText =
+      File(pathToMavenInstall).inputStream().bufferedReader().use { it.readText() }
+    val moshi = Moshi.Builder().addLast(KotlinJsonAdapterFactory()).build()
+    val adapter = moshi.adapter(MavenListDependencyTree::class.java)
+    val dependencyTree = adapter.fromJson(mavenInstallJsonText)
+    return dependencyTree?.mavenListDependencies?.dependencyList?.filter { dep ->
+      bazelQueryDepsNames.contains(
+        omitVersionAndReplaceColonsHyphensPeriods(dep.coord)
+      )
+    } ?: listOf<MavenListDependency>()
+  }
+
+  private fun getNodeValue(tag: String, element: Element): String {
+    val nodeList = element.getElementsByTagName(tag)
+    val node = nodeList.item(0)
+    if (node != null) {
+      if (node.hasChildNodes()) {
+        val child = node.getFirstChild()
+        while (child != null) {
+          if (child.getNodeType() === Node.TEXT_NODE) {
+            return child.getNodeValue()
+          }
+        }
+      }
+    }
+    return ""
+  }
+
+  private fun replaceHttpWithHttps(
+    url: String
+  ): String {
+    var httpsUrl = url
+    if (!url.startsWith("https")) {
+      httpsUrl = url.replace("http", "https")
+    }
+    return httpsUrl
```

Comment:
Maybe another case where removePrefix is helpful:

```suggestion
    return "https${url.removePrefix("https").removePrefix("http")}"
```

Though that's slightly hacky.

Or you could ``replaceFirst("http://", "https://")``.

Reply:
Done.